CaloDiag/src/CaloExample_module.cc

```diff
@@ -368,20 +371,26 @@ namespace mu2e {
               }    		          
            }
  
-           float cryDT(999);
-           if (hit.recoCaloDigis().size()==2)
+           float cryT1(999),cryT2(999),cryT1Err(999),cryT2Err(999);
+           if (hit.recoCaloDigis().size()>1)
            {
-              if (hit.recoCaloDigis().at(0)->SiPMID()%2==0) cryDT = hit.recoCaloDigis().at(1)->time()- hit.recoCaloDigis().at(0)->time();
-              else                                          cryDT = hit.recoCaloDigis().at(0)->time()- hit.recoCaloDigis().at(1)->time();
+              int idx0 = hit.crystalID()%2==0 ? 0 : 1; //hit.recoCaloDigis().at(0)->SiPMID()%2==0 ? 0 : 1;
+              int idx1 = 1-idx0;
```

Comment:
Here is an idea that we can start now and fix other places as targets of opportunity.

Since the beginning I have never liked exposing the idx algebra; I recognize it for what it is - deducing the the two SiPM indices from the Crystal ID.  I hope that during commissioning many new people will contribute to the calo code and I don't want them having to write this indexology correctly every time.

Wrap the idx0 and idx1 algebra in a fully inlined struct with  two data members ( int idx0 and int idx1 ) and do the work in the c'tor.  Or will this degrade optimization in the case that you only need idx0 and not idx1?  There are solutions for that case too.

You can put the new class in DataProducts/inc, which has evolved into low level enum's and indices .

Reply:
I have already a facility - caloIDMapper - that deals with all crystal / SiPM indices. I added a function to deal with this case.